platform/core/src/types/DataSourceConfigurationAPI.ts

```diff
@@ -0,0 +1,67 @@
+export interface BaseDataSourceConfigurationAPIItem {
+  id: string;
+  name: string;
+}
+
+/**
+ * The interface to use to configure an associated data source. Typically an
+ * instance of this interface is associated with a data source that the instance
+ * understands and can alter the data source's configuration.
+ */
+export interface BaseDataSourceConfigurationAPI {
```

Comment:
It makes sense to add a section to our docs to show what is possible (screenshots) and how they can create their own configuration components